modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,541 @@
+        function: Callable,
+        result_schema: Optional[Dict[Hashable, type]] = None,
+    ) -> "ModinDataframe":
+        """
+        Explode data based on the function provided along the specified axis.
+
+        Parameters
+        ----------
+        axis : int or modin.core.dataframe.base.dataframe.Axis
+            The axis to expand over.
+        function : callable
+            The function to use to expand the data. This function should accept one
+            row/column, and return multiple.
+        result_schema : dictionary, optional
+            Mapping from column labels to data types that represents the types of the output dataframe.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the specified axis expanded.
+
+        Notes
+        -----
+        Only one axis can be expanded at a time.
+
+        The user-defined function may increase the number of rows (columns if axis=1),
+        but it should not remove or drop rows.
+        """
+        pass
+
+    @abstractmethod
+    def window(
+        self,
+        axis: Union[int, Axis],
+        reduce_fn: Callable,
+        window_size: int,
+        result_schema: Optional[Dict[Hashable, type]] = None,
+    ) -> "ModinDataframe":
+        """
+        Apply a sliding window operator that acts as a GROUPBY on each window, reducing each window to a single row (column).
+
+        Parameters
+        ----------
+        axis : int or modin.core.dataframe.base.dataframe.Axis
+            The axis to slide over.
+        reduce_fn : callable(rowgroup|colgroup) -> row|col
+            The reduce function to apply over the data.
+        window_size : int
+            The number of row/columns to pass to the function.
+            (The size of the sliding window).
+        result_schema : dictionary, optional
+            Mapping from column labels to data types that represents the types of the output dataframe.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the reduce function applied over windows of the specified
+            axis.
+
+        Notes
+        -----
+        The user-defined reduce function must reduce each window’s column
+        (row if axis=1) down to a single value.
+        """
+        pass
+
+    @abstractmethod
+    def groupby(
+        self,
+        axis: Union[int, Axis],
+        by: Union[str, List[str]],
+        operator: Callable,
+        result_schema: Optional[Dict[Hashable, type]] = None,
+    ) -> "ModinDataframe":
+        """
+        Generate groups based on values in the input column(s) and perform the specified operation on each.
+
+        Parameters
+        ----------
+        axis : int or modin.core.dataframe.base.dataframe.Axis
+            The axis to apply the grouping over.
+        by : string or list of strings
+            One or more column labels to use for grouping.
+        operator : callable
```

Comment:
Does `operator` relates with the dataframe algebra somehow? Maybe `operation` or `function` would a better name for the parameter?

Reply:
In this case, the `operator` should be an algebraic operator, so I think it makes sense to keep it as is.

Reply:
If so, I agree.